src/parver/_version.py

```diff
@@ -313,13 +313,13 @@ class Version:
 
     pre_sep1: Optional[Separator] = attr.ib(default=None, validator=validate_sep)
     pre_sep2: Optional[Separator] = attr.ib(default=None, validator=validate_sep)
-    post_sep1: Optional[Separator] = attr.ib(
+    post_sep1: Union[Separator, UnsetType, None] = attr.ib(
         default=UNSET, validator=validate_sep_or_unset
     )
-    post_sep2: Optional[Separator] = attr.ib(
+    post_sep2: Union[Separator, UnsetType, None] = attr.ib(
         default=UNSET, validator=validate_sep_or_unset
     )
-    dev_sep: Optional[Separator] = attr.ib(
+    dev_sep: Union[Separator, UnsetType, None] = attr.ib(
```

Comment:
This was intentional iirc since `UNSET` is a private type, but is of course wrong, and we already have `UnsetType` in the signature for `Version.replace` so I will accept this change.

Nitpick: can you please change the order to `Union[Separator, None, UnsetType]`

Reply:
having looked at this again, this is the best we can do since it's accurate that the attribute is never UnsetType. 

It's like this (and maybe I should have implemented `__init__` manually at this point...)

```python
class Version:
    post_sep1: Separator | None

    def __init__(post_sep1: Separator | None | UnsetType) -> None:
        if post_sep1 is UNSET:
            self.post_sep1 = None
        else:
            ...
```